dags/iop/parser.py

```diff
@@ -170,27 +189,30 @@ def _get_affiliations(self, article):
         )
         authors = []
         for contrib_type in contrib_types:
-            surname = contrib_type.find("name/surname").text
-            given_names = contrib_type.find("name/given-names").text
-            reffered_id = contrib_type.find("xref[@ref-type='aff']")
-            if reffered_id is None:
+            surname = get_text_value(contrib_type.find("name/surname"))
+            given_names = get_text_value(contrib_type.find("name/given-names"))
+            reffered_ids = [
+                reffered_id
+                for reffered_id in contrib_type.findall("xref[@ref-type='aff']")
+                if reffered_id is not None
+            ]
+            if reffered_ids is None:
                 continue
-            institutions = self._get_institutions(article, reffered_id)
-            countries = self._get_countries(article, reffered_id)
-            affiliations = self._get_affiliation_values(institutions, countries)
-            author = {
-                "surname": surname,
-                "given_names": given_names,
-                "affiliations": affiliations,
-            }
-            authors.append(author)
+            affiliations = self._get_affiliation_values(article, reffered_ids)
+            if "collaboration" not in given_names.lower():
+                author = {
+                    "surname": surname,
+                    "given_names": given_names,
+                    "affiliations": affiliations,
+                }
+                authors.append(author)
         return authors
 
     def _get_date(self, date):
         return datetime.date(
-            day=int(date.find("day").text),
-            month=int(date.find("month").text),
-            year=int(date.find("year").text),
+            day=int(get_text_value(date.find("day"))),
+            month=int(get_text_value(date.find("month"))),
+            year=int(get_text_value(date.find("year"))),
```

Comment:
`get_text_value` if the `key` doesn't exist it will return `None`. Hence the `int` will fail. For these type of cases we should have tests :)

Reply:
Good point, I will write a test when all not required fields are missing